samples/src/Universal/Events/AspectEventArgs.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Reflection;
+using Aspects.Universal.Attributes;
+
+namespace Aspects.Universal.Events
+{
+    public class AspectEventArgs : EventArgs
+    {
+        public object Instance { get; set; }
+        public Type Type { get; set; }
+        public MethodBase Method { get; set; }
+        public Func<object[], object> Target { get; set; }
+        public string Name { get; set; }
+        public object[] Args { get; set; }
+        public Type ReturnType { get; set; }
+        public BaseUniversalWrapperAttribute[] Triggers { get; set; }
+
+        public AspectEventArgs Copy()
+        {
+            return (AspectEventArgs)MemberwiseClone();

```

Comment:
What is the point of copying? Why not readonly?

Reply:
I make a copy to be sure that the user cannot change the values.

Or do you mean something else?

Reply:
I mean user still can change a lot. Almost all fields are byref, so copying makes just another reference. For example user can change arguments. Reference to array allows change individual elements

Reply:
In that case, I think we should just `trust` the user to not break the values. 

So I think that we can remove the copy logic.

Reply:
I agree. But again why not make all properties readonly and args property ```IReadOnlyList<T>``` which is an interface of ```T[]``` ?

Reply:
I've updated to use IReadOnlyList and made setters internal .

Reply:
Looks good!